components/scream/src/share/util/scream_common_physics_functions_impl.hpp

```diff
@@ -384,6 +384,78 @@ void PhysicsFunctions<DeviceT>::calculate_mmr_from_vmr(const MemberType& team,
   });
 }
 
+//+++++++++++PMC
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+void PhysicsFunctions<DeviceT>::lapse_T_for_psl(const ScalarT& T_ground, const ScalarT& p_ground, const ScalarT& phi_ground,
+						ScalarT& lapse, ScalarT& T_ground_tmp )
+{
+  /* 
+    Choose lapse rate and effective ground temperature to use for sea-level pressure calculation.
+    This function should only be used by calculate_psl and is separated from that function solely 
+    to improve specificity of unit testing.
+ */
+
+  using C = scream::physics::Constants<Real>;
+  constexpr Real gravit = C::gravit;
+
+  //Get preliminary surface and sea level temperature to decide on lapse rate
+  auto T_sl = T_ground + 0.0065*phi_ground/gravit; //start by assuming lapse rate is 6.5 K/km
+  T_ground_tmp= T_ground; //make copy b/c may need to modify later
+
+  if (T_ground_tmp<=290.5 && T_sl>290.5){
```

Comment:
I was surprised that these functions are called from inside a loop over columns so - since we're only dealing with surface variables - these functions will only ever be called on scalars... not ScalarT=pack, but an actual scalar. It seems like this might limit these functions' usability (e.g. for writing output). More importantly, since these functions are called on scalars, I'm assuming it's fine to have as many if/else statements as I feel like. Another thing which could be a problem is perhaps I should be declaring things Real instead of ScalarT and not templating on ScalarT?

Reply:
I think it's fine to keep templated on ScalarT. The code should not compile for a pack type, so there should be no risk of sneaking in bugs with unintentional behaviors.

That said, hard coding Real is perfectly fine too.

Reply:
Changed to hard-coded Real in order to be more explicit about what these variables are.